lib/tasks/pact.rake

```diff
@@ -1,5 +1,17 @@
 return if Rails.env.production?
 
 require "pact/tasks"
-require "pact_broker/client/tasks"
 require "pact/tasks/task_helper"
+
+desc "Verify that Collections honours all contracts in the pactfile created by a given branch of gds-api-adapters"
+task "pact:verify:branch", [:branch_name] => :environment do |t, args|
+  abort "Please provide a branch name. eg rake #{t.name}[my_feature_branch]" unless args[:branch_name]
+
+  pact_version = args[:branch_name] == "master" ? args[:branch_name] : "branch-#{args[:branch_name]}"
+
+  ClimateControl.modify GDS_API_PACT_VERSION: pact_version do
+    Pact::TaskHelper.handle_verification_failure do
+      Pact::TaskHelper.execute_pact_verify
+    end
+  end
+end
```

Comment:
I notice in the Publishing API version of this file we've got a bit at the bottom with credentials, do we need that here?

https://github.com/alphagov/publishing-api/blob/a0c59b9c7b8ec48fa4256274abed626567aea87a/lib/tasks/pact.rake#L20-L30

Reply:
I believe the credentials and the definition of the publication task are needed to publish the pactfile to the broker, which is the role of the service consumer. 

In the case of publishing api as it is both a [consumer](https://github.com/alphagov/content-store/blob/c665fa0fb3e513c42187a1f8dd4ff4a5f264fe28/spec/service_consumers/pact_helper.rb#L19) and a [provider](https://github.com/alphagov/publishing-api/blob/a0c59b9c7b8ec48fa4256274abed626567aea87a/spec/service_consumers/pact_helper.rb#L19) it defines the `pact:verify:branch` in its role as a provider,  and it calls `PactBroker::Client::PublicationTask` to publish the contracts it has defined with the content-store.

The publication task for the pact between collections and gds-api-adapters is defined in the [latter](https://github.com/alphagov/gds-api-adapters/blob/3f14140bf8e7170308baf125a161921546d88bef/Rakefile#L19toL29) so it isn't needed here. That's my understanding, I may have got the wrong end of the stick!

Reply:
Ah yes, I think that makes sense. I guess as we roll this out to more apps we may find that we want a common file which includes both, even if it's unused for some apps, but I think this if fine for now.